lib/src/over_react_redux/hooks/use_selector.dart

```diff
@@ -0,0 +1,228 @@
+  bool Function(TValue tNextValue, TValue tPrevValue) equalityFn,
+]) {
+  ReactInteropValue jsSelector(ReactInteropValue jsState) => wrapInteropValue(selector(unwrapInteropValue(jsState)));
+  _JsReduxStateEqualityFn jsEqualityFn = equalityFn == null
+      ? null
+      : allowInterop((nextJsValue, prevJsValue) =>
+          equalityFn(unwrapInteropValue(nextJsValue), unwrapInteropValue(prevJsValue)));
+
+  return unwrapInteropValue<TValue>(_jsUseSelector(allowInterop(jsSelector), jsEqualityFn));
+}
+
+@JS('ReactRedux.useSelector')
+external ReactInteropValue _jsUseSelector(_JsSelectorFn selector, [_JsReduxStateEqualityFn equalityFn]);
+
+// ----------------------------------------------------
+//  createSelectorHook
+// ----------------------------------------------------
+
+/// Returns a function that can be named and used as a replacement for [useSelector] when you want to
+/// use custom [context].
+///
+/// This is useful if you're building a complex reusable component, and you don't want your `Store` to collide
+/// with any Redux `Store` your consumers' applications might use.
+///
+/// See the [react-redux JS documentation](https://react-redux.js.org/api/hooks#custom-context) for more details.
+///
+/// ### Custom Context Example
+///
+/// ```dart
+/// import 'package:over_react/over_react.dart';
+/// import 'package:over_react/over_react_redux.dart';
+/// import 'package:redux/redux.dart';
+///
+/// // ------------------------------------
+/// //  1. Declare the custom context
+/// // ------------------------------------
+/// final MyContext = createContext();
+/// final useSelector = createSelectorHook<MyState>(MyContext);
+/// final myStore = Store(myReducer);
+///
+/// // ------------------------------------
+/// //  2. Create a function component that
+/// //  uses the shadowed `useSelector`.
+/// // ------------------------------------
+/// mixin MyComponentProps on UiProps {}
+///
+/// UiFactory<MyComponentProps> MyComponent = uiFunction(
+///   (props) {
+///     final count = useSelector((state) => state.count);
+///
+///     return Dom.div()('The current count is $count');
+///   },
+///   $MyComponentConfig, // ignore: undefined_identifier
+/// );
+///
+/// // ------------------------------------
+/// //  3. Render the function component
+/// //  nested within the ReduxProvider
+/// //  that is wired up to the
+/// //  custom context / store.
+/// // ------------------------------------
+/// main() {
+///   final app = (ReduxProvider()
+///     ..context = MyContext
+///     ..store = myStore
+///   )(
+///     MyComponent()(),
+///   );
+///
+///   react_dom.render(app, querySelector('#id_of_mount_node'));
+/// }
+/// ```
+_SelectorFnHook<TReduxState> createSelectorHook<TReduxState>([Context context]) {
+  final jsHook = _jsCreateSelectorHook(context?.jsThis ?? JsReactRedux.ReactReduxContext);
+  // Using a var instead of a named function here to avoid having to redeclare the type on the LHS
```

Comment:
Can we make this a named function? I think the benefit of getting better stack traces is worth having to redeclare the types.